src/System.Runtime.Numerics/tests/BigInteger/ctor.cs

```diff
@@ -924,7 +924,7 @@ public static void RunCtorByteArrayTests()
             // ctor(byte[]): array is large
             for (int i = 0; i < s_samples; i++)
             {
-                tempByteArray = new byte[s_random.Next(16384, 2097152)];
+                tempByteArray = new byte[s_random.Next(16384, 262144)];
```

Comment:
I would say if this needs to be changed to get the tests to pass, the algorithm is a regression from its current implementation, no? As for should the stack size increase or a move to fixed, I don't know enough to comment on the right approach.